File: backend/src/lib/estado-civil-voluntario.ts
import { EstadoCivilVoluntario } from "@prisma/client";

export const ESTADOS_CIVIS_VOLUNTARIO = [
  { valor: "CASADO" as const, rotulo: "Casado(a)" },
  { valor: "DIVORCIADO" as const, rotulo: "Divorciado(a)" },
  { valor: "SEPARADO" as const, rotulo: "Separado(a)" },
  { valor: "SOLTEIRO" as const, rotulo: "Solteiro(a)" },
  { valor: "VIUVO" as const, rotulo: "Viúvo(a)" },
] as const;

export const ESTADO_CIVIL_VOLUNTARIO_VALUES = ESTADOS_CIVIS_VOLUNTARIO.map(
  (e) => e.valor,
) as [EstadoCivilVoluntario, ...EstadoCivilVoluntario[]];

const porValor = new Map(ESTADOS_CIVIS_VOLUNTARIO.map((e) => [e.valor, e]));

export function rotuloEstadoCivilVoluntario(
  valor: EstadoCivilVoluntario,
): string {
  return porValor.get(valor)?.rotulo ?? valor;
}

File: mobile/lib/constants/estado_civil_voluntario.dart
/// Estado civil do voluntário (valores da API / enum Prisma).
abstract final class EstadoCivilVoluntario {
  static const casado = 'CASADO';
  static const divorciado = 'DIVORCIADO';
  static const separado = 'SEPARADO';
  static const solteiro = 'SOLTEIRO';
  static const viuvo = 'VIUVO';

  static const opcoes = <String, String>{
    casado: 'Casado(a)',
    divorciado: 'Divorciado(a)',
    separado: 'Separado(a)',
    solteiro: 'Solteiro(a)',
    viuvo: 'Viúvo(a)',
  };

  static String? rotulo(String? valor) {
    if (valor == null || valor.isEmpty) return null;
    return opcoes[valor];
  }
}
